core/server/master/src/main/java/alluxio/master/file/replication/ReplicationChecker.java

```diff
@@ -200,6 +216,12 @@ public void close() {
   private void checkMisreplicated(Set<Long> inodes, ReplicationHandler handler)
       throws InterruptedException {
     for (long inodeId : inodes) {
+      if (mActiveJobToInodeID.size() >= MAX_ACTIVE_JOBS) {
```

Comment:
one thing i am concerned about, if we have 20k files that need replication, the first 10k files are missing in the ufs. This implementation will keep trying those 10k and never get to the second half?

Reply:
Look at mActiveInodeIDS.contains(inodeId) check.. 

On first run of this, it's going to schedule the first 10k files. On the 2nd run, if 3k of them are completed, it will remove the inodeids that are completed off the list and schedule new ones. (though while saying this, I noticed a bug in my code)